src/tools.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { test, expect } from "vitest";
+import { transpose } from "./tools";
+
+test("transpose", () => {
+  const boardState = [
+    [0, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 0, 0],
+    [0, 2, 1, 1, 2, 0, 2],
+    [0, 1, 1, 2, 1, 1, 1],
+    [0, 2, 2, 1, 1, 2, 2],
+  ];
+  const tBoardState = [
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 2, 2, 1, 2],
+    [0, 0, 1, 1, 1, 2],
+    [0, 0, 0, 1, 2, 1],
+    [1, 2, 2, 2, 1, 1],
+    [0, 0, 0, 0, 1, 2],
+    [0, 0, 0, 2, 1, 2],
+  ];
+
+  expect(transpose(boardState)).toEqual(tBoardState);
+});
```

Comment:
Nice ! Some tests could be added (for edge cases like a lines of differents sizes for example) but this is for later.